media-source/mediasource-preload.html

```diff
@@ -0,0 +1,84 @@
+<!DOCTYPE html>
+<!-- Copyright © 2016 Chromium authors and World Wide Web Consortium, (Massachusetts Institute of Technology, ERCIM, Keio University, Beihang). -->
+<html>
+    <head>
+        <title>Various MediaSource HTMLMediaElement preload tests.</title>
+        <link rel="author" title="Matthew Wolenetz" href="mailto:wolenetz@chromium.org"/>
+        <script src="/resources/testharness.js"></script>
+        <script src="/resources/testharnessreport.js"></script>
+    </head>
+    <body>
+        <script>
+            function attachWithPreloadTest(preload)
+            {
+                async_test(function(test)
+                {
+                    var video = document.createElement("video");
+                    var mediaSource = new MediaSource();
+                    var mediaSourceURL = URL.createObjectURL(mediaSource);
+
+                    video.preload = preload;
+                    document.body.appendChild(video);
+                    test.add_cleanup(function() {
+                        document.body.removeChild(video);
+                        URL.revokeObjectURL(mediaSourceURL);
+                    });
+
+                    var listener = test.step_func(function(event) { test.done(); });
+                    mediaSource.addEventListener("sourceopen", listener);
+                    video.src = mediaSourceURL;
+                }, "sourceopen occurs with element preload=" + preload);
+            }
+
+            attachWithPreloadTest("auto");
+            attachWithPreloadTest("metadata");
+            attachWithPreloadTest("none");
+
+            function errorWithPreloadTest(preload, bogusURLStyle)
+            {
+                var testURLdescription;
+                var mediaSource = new MediaSource();
+                var bogusURL;
+
+                if (bogusURLStyle == "revoked") {
+                    testURLdescription = "revoked MediaSource object URL";
```

Comment:
Uh, I guess "blob:a" is the tricky case. I think it'd actually be fine to merge "corrupted" and "blob:a" into a single case, since it would be hitting the same code path (not even a well-formed blob URL).